Convert all cron jobs to async/await

nwalters512 opened the pull request:
Almost all cron jobs were already using async/await internally and just wrapped in `util.callbackify`. This PR converts the last few and changes `cron/index.js` to expect an native async `run` function.

nwalters512 commented on apps/prairielearn/src/cron/autoFinishExams.ts:
```diff
@@ -15,46 +15,41 @@ import { config } from '../lib/config';
-        const overrideGradeRate = true;
-        // We don't have a client fingerprint ID, so pass null.
-        const clientFingerprintId = null;
-        assessment.gradeAssessmentInstance(
```
This was the last usage of the non-async version of this function, so I dropped the `gradeAssessmentInstanceAsync` wrapper and made this function itself async 🎉

nwalters512 commented on apps/prairielearn/src/cron/autoFinishExams.ts:
```diff
@@ -15,46 +15,41 @@ import { config } from '../lib/config';
+      );
+    } catch (err) {
+      logger.error('Error finishing exam', error.addData(err, { examItem }));
+      Sentry.captureException(err, {
```
I figure it can't hurt to capture failures in Sentry?

jonatanschroeder commented on apps/prairielearn/src/cron/autoFinishExams.ts:
```diff
@@ -15,46 +15,41 @@ import { config } from '../lib/config';
+    // We don't have a client fingerprint ID, so pass null.
+    const clientFingerprintId = null;
+    try {
+      assessment.gradeAssessmentInstance(
```
Missing await?

jonatanschroeder reviewed (commented):
Won't approve because I believe the missing await is a bug. In any case: I imagine you are minimizing the changes (so, for example, not converting queries to use Zod), but would it make sense to at least convert the exports to ESM, and maybe add a `@ts-check`?

nwalters512 commented:
@jonatanschroeder indeed, I was trying to minimize changes. In 213794f / f829252, I refactored to ESM imports/exports and added `// @ts-check` comments.
